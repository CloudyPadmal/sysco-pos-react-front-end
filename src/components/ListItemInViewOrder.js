import React, { Component } from "react";
import ItemCounterInListItem from "./ItemCounterInListItem";
import PropTypes from "prop-types";
import { connect } from "react-redux";

class ListItemInViewOrder extends Component {

    render() {
        let { productID, productTitle, description, quantity, price } = this.props.singleItem;
        return (
            <li className="list-group-item d-flex justify-content-right"
                style={{ margin: "0px", padding: "0px", width: "100%" }}>
                <div className="row" style={{ margin: "0px", width: "100%" }}>
                    <div className="col-8 card">
                        <div className="card-body" style={{ padding: "5px 0px" }}>
                            <h6 className="card-title d-flex justify-content-right">
                                {productTitle}</h6>
                            <p className="card-subtitle text-muted"
                                style={{ textAlign: "left" }}>{description}</p>
                        </div>
                    </div>
                    <div className="col-2 card">
                        <div className="card-body d-flex align-items-center"
                            style={{ padding: "5px 0px" }}>
                            <ItemCounterInListItem
                                name={productID}
                                count={this.props.itemQuantity[productID]}
                                min={1}
                                max={this.props.clonedItemQuantity[productID] === undefined
                                    ? 0
                                    : this.props.clonedItemQuantity[productID]
                                    + this.props.itemsList.find(
                                        I => I.productID === productID).quantity
                                } />
                        </div>
                    </div>
                    <div className="col-2 card text-right">
                        <p className="card-text" style={{ margin: "5px 0 0 0" }}>
                            @ {price.toFixed(2)}
                        </p>
                        <p className="card-link" style={{ margin: "0px" }}>
                            <b>Rs. {(quantity * price).toFixed(2)}</b>
                        </p>
                    </div>
                </div>
            </li>
        );
    }
}

ListItemInViewOrder.propTypes = {
    passKey: PropTypes.string.isRequired,
    itemQuantity: PropTypes.object.isRequired
};

const mapStateToProps = (state) => ({
    passKey: state.uac.passKey,
    itemQuantity: state.ord.itemQuantity,
    clonedItemQuantity: state.ord.clonedItemQuantity,
    viewItemList: state.ord.viewItemList,
    itemsList: state.ord.itemsList,
    currentOrder: state.ord.currentOrder
});

export default connect(mapStateToProps, null)(ListItemInViewOrder);