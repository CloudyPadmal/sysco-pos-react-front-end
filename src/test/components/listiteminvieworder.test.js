import React from 'react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { mount } from '../enzyme';
import { BrowserRouter as Router } from 'react-router-dom';

import ListItemInViewOrder from '../../components/ListItemInViewOrder';

const middlewares = [thunk];

describe('<ListItemInViewOrder /> component renders', () => {

    const singleItem = {
        productID: 'A',
        productTitle: 'AAA',
        description: 'A aasasas',
        quantity: 100,
        price: 25
    }

    const initialState = {
        uac: { passKey: '' },
        ord: {
            itemQuantity: {},
            clonedItemQuantity: {},
            itemsList: [{
                productID: 'A',
                quantity: 1
            }],
            currentOrder: {}
        },
    };
    const mockStore = configureStore(middlewares);
    let store;

    beforeEach(() => {
        store = mockStore(initialState);
    })

    it('renders', () => {
        const wrapper = mount(
            <Provider store={store}>
                <Router>
                    <ListItemInViewOrder singleItem={singleItem} />
                </Router>
            </Provider>
        );
        expect(wrapper).toMatchSnapshot();
    });
});