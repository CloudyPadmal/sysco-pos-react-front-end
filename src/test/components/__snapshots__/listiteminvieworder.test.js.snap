// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItemInViewOrder /> component renders renders 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(ListItemInViewOrder)
        singleItem={
          Object {
            "description": "A aasasas",
            "price": 25,
            "productID": "A",
            "productTitle": "AAA",
            "quantity": 100,
          }
        }
      >
        <ListItemInViewOrder
          clonedItemQuantity={Object {}}
          currentOrder={Object {}}
          dispatch={[Function]}
          itemQuantity={Object {}}
          itemsList={
            Array [
              Object {
                "productID": "A",
                "quantity": 1,
              },
            ]
          }
          passKey=""
          singleItem={
            Object {
              "description": "A aasasas",
              "price": 25,
              "productID": "A",
              "productTitle": "AAA",
              "quantity": 100,
            }
          }
        >
          <li
            className="list-group-item d-flex justify-content-right"
            style={
              Object {
                "margin": "0px",
                "padding": "0px",
                "width": "100%",
              }
            }
          >
            <div
              className="row"
              style={
                Object {
                  "margin": "0px",
                  "width": "100%",
                }
              }
            >
              <div
                className="col-8 card"
              >
                <div
                  className="card-body"
                  style={
                    Object {
                      "padding": "5px 0px",
                    }
                  }
                >
                  <h6
                    className="card-title d-flex justify-content-right"
                  >
                    AAA
                  </h6>
                  <p
                    className="card-subtitle text-muted"
                    style={
                      Object {
                        "textAlign": "left",
                      }
                    }
                  >
                    A aasasas
                  </p>
                </div>
              </div>
              <div
                className="col-2 card"
              >
                <div
                  className="card-body d-flex align-items-center"
                  style={
                    Object {
                      "padding": "5px 0px",
                    }
                  }
                >
                  <Connect(ItemCounterInListItem)
                    max={0}
                    min={1}
                    name="A"
                  >
                    <ItemCounterInListItem
                      dispatchQuickDeleteThisItemFromQuantity={[Function]}
                      dispatchQuickIndecrementItemFromQuantity={[Function]}
                      itemsList={
                        Array [
                          Object {
                            "productID": "A",
                            "quantity": 1,
                          },
                        ]
                      }
                      max={0}
                      min={1}
                      name="A"
                      passKey=""
                    >
                      <div
                        aria-label="Toolbar with button groups"
                        className="btn-toolbar"
                        role="toolbar"
                      >
                        <div
                          aria-label="Item-Group"
                          className="btn-group"
                          role="group"
                        >
                          <button
                            className="btn btn-secondary"
                            onClick={[Function]}
                            type="button"
                          >
                            +
                          </button>
                          <div
                            style={
                              Object {
                                "textAlign": "center",
                                "width": "3em",
                              }
                            }
                          >
                            <span
                              className="input-group-text"
                              style={
                                Object {
                                  "borderRadius": "0px",
                                  "display": "inline-block",
                                  "height": "100%",
                                  "width": "100%",
                                }
                              }
                            />
                          </div>
                          <button
                            className="btn btn-secondary"
                            onClick={[Function]}
                            type="button"
                          >
                            âˆ’
                          </button>
                        </div>
                        <div
                          aria-label="Delete-Group"
                          className="btn-group"
                          role="group"
                          style={
                            Object {
                              "marginLeft": "10px",
                            }
                          }
                        >
                          <button
                            className="btn btn-danger"
                            onClick={[Function]}
                            type="button"
                          >
                            <i
                              className="fas fa-trash"
                            />
                          </button>
                        </div>
                      </div>
                    </ItemCounterInListItem>
                  </Connect(ItemCounterInListItem)>
                </div>
              </div>
              <div
                className="col-2 card text-right"
              >
                <p
                  className="card-text"
                  style={
                    Object {
                      "margin": "5px 0 0 0",
                    }
                  }
                >
                  @ 
                  25.00
                </p>
                <p
                  className="card-link"
                  style={
                    Object {
                      "margin": "0px",
                    }
                  }
                >
                  <b>
                    Rs. 
                    2500.00
                  </b>
                </p>
              </div>
            </div>
          </li>
        </ListItemInViewOrder>
      </Connect(ListItemInViewOrder)>
    </Router>
  </BrowserRouter>
</Provider>
`;
